src/librustc/middle/expr_use_visitor.rs

```diff
@@ -1283,10 +1283,11 @@ impl<'d,'t,'tcx,TYPER:mc::Typer<'tcx>> ExprUseVisitor<'d,'t,'tcx,TYPER> {
     }
 }
 
-fn copy_or_move<'tcx>(typer: &mc::Typer<'tcx>,
-                      cmt: &mc::cmt<'tcx>,
-                      move_reason: MoveReason)
-                      -> ConsumeMode
+fn copy_or_move<'tcx,T>(typer: &T,
+                        cmt: &mc::cmt<'tcx>,
+                        move_reason: MoveReason)
+                        -> ConsumeMode
+    where T: mc::Typer<'tcx>
```

Comment:
Oh, I see, Typer is no longer object-safe. I think I'd prefer that Typer stay object-safe; pity we don't have a good solution for passing FnOnce() closurs as objects yet.